src/app/task-create/task-create.component.ts

```diff
@@ -24,16 +27,22 @@ export class TaskCreateComponent implements OnInit {
     });
   }
 
-  ngOnInit(): void {}
-
+  ngOnInit(): void { }
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
   onCreateTask() {
     if (this.taskForm.valid) {
       const newTask: Task = {
         ...this.taskForm.value,
         createdOn: new Date()
       };
-      this.taskService.addTask(newTask);
-      this.taskForm.reset({ status: 'pending' }); 
+      this.taskService.createTask(newTask).pipe(takeUntil(this.destroy$)).subscribe(() => {
+        this.taskForm.reset({ status: 'pending' });
```

Comment:
if you're navigating away from the page, then the taskForm will reset itself because the component is destroyed